File: app/src/main/java/net/claztec/simplegithub/api/GithubApi.java
package net.claztec.simplegithub.api;

import net.claztec.simplegithub.api.model.GithubRepo;
import net.claztec.simplegithub.api.model.RepoSearchResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface GithubApi {

    @GET("search/repositories")
    Call<RepoSearchResponse> searchRepository(@Query("q") String query);

    @GET("repos/{owner}/{name}")
    Call<GithubRepo> getRepository(@Path("owner") String ownerLogin, @Path("name") String repoName);
}

File: app/src/main/kotlin/net/claztec/simplegithub/api/GithubApi.kt
package net.claztec.simplegithub.api

import io.reactivex.Observable
import net.claztec.simplegithub.api.model.GithubRepo
import net.claztec.simplegithub.api.model.RepoSearchResponse
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface GithubApi {

    @GET("search/repositories")
    fun searchRepository(@Query("q") query: String): Observable<RepoSearchResponse>

    @GET("repos/{owner}/{name}")
    fun getRepository(@Path("owner") ownerLogin: String, @Path("name") repoName: String): Observable<GithubRepo>
}
